Update Linux support to 0.0.13

sarnex opened the pull request:
Code heavily based on @Francesco149's original PR, with line numbers updated.

Signed-off-by: Nick Sarnie commendsarnex@gmail.com

His original instructions are below, with slight modification:
**Make sure you have at least node v5**

```


cd
mkdir discord
cd discord

# Install discord in /opt/
wget https://discordapp.com/api/download/canary?platform=linux&format=tar.gz
tar -xzvf discord-canary-0.0.3.tar.gz
sudo cp -r DiscordCanary /opt/
sudo ln -s /opt/DiscordCanary/DiscordCanary /usr/bin/discord

# Install BetterDiscord
git clone https://github.com/sarnex/BetterDiscordApp.git
mv BetterDiscordApp BetterDiscord
cp BetterDiscord/Installers/Node/* .
cp BetterDiscord/splice .
npm install asar wrench
sudo ./install.sh
```

DaFri-Nochiterov commented on NodeInstaller/index.js:
```diff
@@ -1,248 +1,255 @@
+			_functionCallSplice = 507;
+			_functionSplice = 580;
+			_discordPath = "/opt/DiscordCanary/resources";
+			// not sure where the .deb install, and most people are going to 
```
AFAIK `*.deb` file installs in `/usr/share/discord-canary`

sarnex replied:
The original required the user to install in /opt, and not everyone is on Debian or Ubuntu, so I'm inclined to leave that for now.

DaFri-Nochiterov commented:
```bash
cd

# Move dir if exists
oldTime=$(date +"%H-%M-%S.%d-%m-%y")
mv discord "discord.${oldTime}.old"

# And.. Create new!
mkdir discord
cd discord

# Install discord in /opt/
wget "https://discordapp.com/api/download/canary?platform=linux&format=tar.gz" -O discord-canary-0.0.3.tar.gz
tar -xzvf discord-canary-0.0.3.tar.gz
sudo cp -r DiscordCanary /opt/
sudo ln -s /opt/DiscordCanary/DiscordCanary /usr/bin/discord

# Install BetterDiscord
git clone https://github.com/sarnex/BetterDiscordApp.git
mv BetterDiscordApp BetterDiscord
cp BetterDiscord/NodeInstaller/* .
cp BetterDiscord/splice .
npm install asar wrench
sudo ./install.sh
```

Francesco149 commented:
ah thanks for updating my pull request, was too busy to work on it

mkody commented:
Works fine for me!

I needed to change the `_discordPath` in the `NodeInstaller/index.js` file since the [AUR package for Discord-Canary](https://aur.archlinux.org/packages/discord-canary/) uses `/opt/discord-canary/`. 
But it looks like this path is not consistent between distributions. But for Arch, someone might just add a patch during installation if they make a AUR package, if needed :+1:

sarnex commented:
Still works with no change on version 0.4

DaFri-Nochiterov commented:
Not works on newest 0.0.6 version :(

sarnex commented:
@DaFri-Nochiterov should be fixed now

DaFri-Nochiterov commented:
@sarnex, yes, this works. Thanks.

DaFri-Nochiterov commented:
Again. Not works on 0.0.7 version.

Now with `Exception`:
```javascript
Uncaught Exception:
SyntaxError: Unexpected token ;
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:387:25)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Object.<anonymous> (/opt/DiscordCanary/resources/electron.asar/browser/init.js:184:8)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
```

sarnex commented:
@DaFri-Nochiterov fixed again, hopefully they leave index.js alone. If not, if it happens again, I'll look for a better solution than hardcoding line numbers, as is done in the official installer.

mkody commented:
@sarnex You should definitely look for a better solution, because I'm pretty sure the index.js will change a lot in the near future.

sarnex commented:
I've rewritten the linux installer to not use hardcoded line numbers. It should not break now unless they remove the two lines I'm searching for, which I don't expect.

mkody commented:
And it works fine on just-released `0.0.8`! 👍

DaFri-Nochiterov commented:
Emoticons no longer works in this version

sarnex commented:
No idea why, installer doesn't seem to have been modified. Someone else should investigate.

mkody commented:
Huh? It works for me. Running from commit 6be4da2197ea57025f653a07903fa01764110c70

sarnex commented:
Emotes work now, it was a BetterDIscord bug unrelated to these changes.

simoniz0r commented:
I have tried installing this multiple times, but I keep getting this error after running `sudo ./install.sh`

`simonizor@T00L-Xubuntu:~/Downloads/discord$ sudo ./install.sh 
/home/simonizor/Downloads/discord/index.js:128
 fs.exists(process.env.HOME + _prefsPath + emotes_twitch_global, (exists) => {
                                                                           ^
SyntaxError: Unexpected token >
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3
simonizor@T00L-Xubuntu:~/Downloads/discord$
`

Any help would be appreciated.

chowder908 commented:
I'm getting the same error as simoniz0r.

sarnex commented:
@simoniz0r @chowder908 

I've rebased the pull request based on the lastest BetterDiscord, and I tried the install from scratch and it worked. I've also updated the original instructions to fit the current directory structure. 

Let me know if there's still an issue.

chowder908 commented:
sudo: ./install.sh: command not found

Now telling me this.

chowder908 commented:
Is there anyway to just manually install it like copying the files to the places they have to be?

chowder908 commented:
@sarnex  sudo: ./install.sh: command not found
Now telling me this.
Is there anyway to just manually install it like copying the files to the places they have to be?

sarnex commented:
@chowder908 you have to chmod +x install.sh

chowder908 commented:
@sarnex 
```
sudo chmod +x ./install.sh
chmod: cannot access ‘./install.sh’: No such file or directory

```

sarnex commented:
@chowder908 did you follow the instructions above? 
````
git clone https://github.com/sarnex/BetterDiscordApp.git
mv BetterDiscordApp BetterDiscord
cp BetterDiscord/Installers/Node/* .
cp BetterDiscord/splice .
npm install asar wrench
sudo ./install.sh
````

chowder908 commented:
@sarnex line by line

simoniz0r commented:
```
simonizor@T00L-Xubuntu:~/Downloads/discord$ sudo ./install.sh 

/home/simonizor/Downloads/discord/index.js:126
 fs.exists(process.env.HOME + _prefsPath + emotes_twitch_global, (exists) => {
                                                                           ^
SyntaxError: Unexpected token >
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3
```
I'm getting pretty much the same error after deleting everything and cloning your PR after these changes.

chowder908 commented:
Yeah @simoniz0r i redid it and same error.

sarnex commented:
@simoniz0r Can you send me the output of node -v? I think it's some weird library or node version issue.

simoniz0r commented:
```
simonizor@T00L-Xubuntu:~/Downloads/discord$ node -v
v0.10.25
```

chowder908 commented:
@sarnex I have v0.10.29

sarnex commented:
That version is from around 2014. See if you can update it. I'm on 5.10.1

simoniz0r commented:
@sarnex Thank you, that worked!

@chowder908 In case you need help updating node, this is how I did it:
```
sudo npm cache clean -f
sudo npm install -g n
sudo n stable

sudo ln -sf /usr/local/n/versions/node/<VERSION>/bin/node /usr/bin/node 
```

chowder908 commented:
@sarnex @simoniz0r i'm getting sudo: npm: command not found

sarnex commented:
@simoniz0r cool, glad it worked. I'll add it to the original post

Tsunamical commented on Installers/Node/index.js:
```diff
@@ -99,8 +101,18 @@ function install() {
-                    data.splice(_importSplice, 0, 'var _betterDiscord = require(\'betterdiscord\');\n');
-                    data.splice(_functionCallSplice, 0, splice);
+                    
+                    if(_os == "linux") {
```
darwin needs this now too

Tsunamical replied:
Actually, we can just remove the if clause here, #336 has it by default. Should be functional across the board.

carlmiller307 commented:
I installed this on my copy of DiscordCanary, which runs on Ubuntu 16.04 x64, and the load process hangs on FrankerFaceZ emote data.

mkody commented:
@carlmiller307 Your problem looks unrelated to this pull request. Here we're talking about the installation script. If it says that FrankerFaceZ loads, that mean it's installed (but has a bug).
You should try to [open a new issue](https://github.com/Jiiks/BetterDiscordApp/issues/new) for that.

GhettoGirl commented:
I'm getting this error in the Discord app 0.0.8


>(program):1 Uncaught TypeError: Cannot read property 'appendChild' of undefinedelectron.ipcRenderer.on @ /opt/discord-canary/resources/electron.asar/renderer/init.js:44
>(program):1 Uncaught TypeError: Cannot set property 'innerHTML' of nullelectron.ipcRenderer.on @ /opt/discord-canary/resources/electron.asar/renderer/init.js:44
>(program):1 Uncaught TypeError: Cannot set property 'value' of nullelectron.ipcRenderer.on @ /opt/discord-canary/resources/electron.asar/renderer/init.js:44
>(program):1 Uncaught TypeError: Cannot set property 'max' of nullelectron.ipcRenderer.on @ /opt/discord-canary/resources/electron.asar/renderer/init.js:44


Node v6.4.0
Arch Linux

I applied this pull request on top of **master** (48d4a58102dcdbc1ce888b322df2f7aa834eb695)

mkody commented:
@GhettoGirl Hmm... Works fine for me, I'm using the `discord-canary` AUR package.
```
yaourt -S discord-canary
# I've started Discord at least once and then quit before installing BetterDiscord
mkdir BDA && cd BDA
git clone https://github.com/Jiiks/BetterDiscordApp.git && cd BetterDiscordApp
git remote add sarnex https://github.com/sarnex/BetterDiscordApp.git
git pull sarnex master
cd .. && mv BetterDiscordApp BetterDiscord
cp BetterDiscord/Installers/Node/* BetterDiscord/splice .
npm install asar wrench
sudo ./install.sh
```

GhettoGirl commented:
@mkody I restarted Discord several times, now it works. It may be related to this upstream bug https://github.com/crmarsh/discord-linux-bugs/issues/81. But this error is new, I'm gonna report this in the upstream issue too.

jamesguessis commented:
do plugins and themes work with this?

mkody commented:
@jamesguessis Of course, yes. What's the point if they don't work? :smile:

jamesguessis commented:
where would the respective folders be located then?

mkody commented:
Inside `~/.config/BetterDiscord`
![1475095264](https://cloud.githubusercontent.com/assets/843787/18931483/ae35492a-85cc-11e6-92c8-e94b4f21b5cd.png)

GhettoGirl commented:
Discord Canary 0.0.9 released and BetterDiscord is no longer compatible with it. Says `_betterDiscord` is undefined. Just want to let you know.

I'm staying at 0.0.8 for a while.

sarnex commented:
Sorry for the delay, I've been busy lately. I've updated the PR for 0.0.9. 

It works for me, please let me know if it works.

Thanks.

GhettoGirl commented:
It works great. Thanks for the quick update. :+1:

mkody commented:
Can confirm it works. :+1:

simoniz0r commented:
Thanks for the update.  I was having some issues with the install script not creating a directory properly before and also the crash above with 0.0.9.  Both of those issues were resolved with this update.  The install went as expected, and no crash on loading 0.0.9.

sarnex commented:
You Rock!
Cancel That
I'm the greatest

sarnex commented:
Works with 0.0.10 without changes.

mkody commented:
Works fine with 0.0.11 too.

sarnex commented:
@mkody thanks for testing

mkody commented:
Aaaaand, works fine with 0.0.12!

sarnex commented:
Tubular

On Tue, Jan 3, 2017 at 3:48 PM, Kody <notifications@github.com> wrote:

> Aaaaand, works fine with 0.0.12!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Jiiks/BetterDiscordApp/pull/288#issuecomment-270220583>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvNDLkCHf6ShbXr_KFfMy3ZrNBaQaZpks5rOrQygaJpZM4INLQo>
> .
>

magiruuvelvet commented:
Still works with 0.0.13

TheDoctorsLife commented:
Any possibility of getting this merged into master? It seems this is pretty solid and plenty of people are reporting it still working with the latest version of Discord Linux.

sarnex commented:
I don't think Jiiks cares.

On Wed, Jan 4, 2017 at 8:59 AM, Benjamin Hoffmeyer <notifications@github.com
> wrote:

> Any possibility of getting this merged into master? It seems this is
> pretty solid and plenty of people are reporting it still working with the
> latest version of Discord Linux.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Jiiks/BetterDiscordApp/pull/288#issuecomment-270376386>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvNDNsetsjP2TKs64gI996vl-RtTm7Jks5rO6W9gaJpZM4INLQo>
> .
>

ddlr commented:
I can also confirm it works with 0.0.13, though oddly enough, the BetterDiscord tab doesn’t appear in the User Settings.

mkody commented:
Yup @ddlr, but it's broken on macOS too (at least). Discord recently pushed a background update that broke BD.

ddlr commented:
@mkody Ah, good to know that I’m not the only one.

magiruuvelvet commented:
There is a temporary workaround for this. The file just needs to be compied to the plugin path, it works automagically. **Don't actually enable this plugin in the BetterDiscord settings** https://gist.github.com/noodlebox/94b26b4ded85a0ded48197333aad290b

simoniz0r commented:
It seems like BetterDiscord updated to v0.2.82... any chance of this PR getting updated also?  BetterDiscord won't load anymore after the update.

http://i.imgur.com/HMTkzIc.png

MelDone commented:
Jup, same error :( 
![screenshot from 2017-01-09 14-20-30](https://cloud.githubusercontent.com/assets/23008210/21767879/d6e056b2-d676-11e6-8bb4-7367f3fd37f5.png)

mkody commented:
I've sent two PR #435 and #436 to fix the latest changes.

So here's what I need to do (_as of right now_) to make it run:
```sh
# Start Discord at least once and then quit before installing BetterDiscord
# Let's create a work dir
mkdir BDA
cd BDA
# Clone the upstream repo to "BetterDiscord"
git clone https://github.com/Jiiks/BetterDiscordApp.git BetterDiscord
cd BetterDiscordApp
# Now we're going to merge the PRs manually
git remote add mkody https://github.com/mkody/BetterDiscordApp.git
git remote add sarnex https://github.com/sarnex/BetterDiscordApp.git
git merge mkody/fix-bdStorage-path
git merge mkody/fix-version-config
git merge sarnex/master
# Now let's go back to our work dir and get the dependencies
cd ..
cp BetterDiscord/Installers/Node/* BetterDiscord/splice .
npm install asar wrench
# And now we can run the installer
# (you might need to use sudo)
bash install.sh
```

### Important note regarding recent changes
Plugins needs to be updated to the new storage system. Please remove them in the `~/.config/plugins` folder before launching BetterDiscord, get the latest version and add them progressively to check if they work.

sarnex commented:
Sorry about that, I just rebased. Let me know if the rebased version works.

On Mon, Jan 9, 2017 at 12:58 PM, Kody <notifications@github.com> wrote:

> I've sent two PR #435 <https://github.com/Jiiks/BetterDiscordApp/pull/435>
> and #436 <https://github.com/Jiiks/BetterDiscordApp/pull/436> to fix the
> latest changes.
>
> So here's what I need to do (*as of right now*) to make it run:
>
> # Start Discord at least once and then quit before installing BetterDiscord# Let's create a work dir
> mkdir BDAcd BDA# Clone the upstream repo to "BetterDiscord"
> git clone https://github.com/Jiiks/BetterDiscordApp.git BetterDiscordcd BetterDiscordApp# Now we're going to merge the PRs manually
> git remote add mkody https://github.com/mkody/BetterDiscordApp.git
> git remote add sarnex https://github.com/sarnex/BetterDiscordApp.git
> git merge mkody/fix-bdStorage-path
> git merge mkody/fix-version-config
> git merge sarnex/master# Now let's go back to our work dir and get the dependenciescd ..
> cp BetterDiscord/Installers/Node/* BetterDiscord/splice .
> npm install asar wrench# And now we can run the installer# (you might need to use sudo)
> bash install.sh
>
> Important note regarding recent changes
>
> Plugins needs to be updated to the new storage system. Please remove them
> in the ~/.config/plugins folder before launching BetterDiscord, get the
> latest version and add them progressively to check if they work.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Jiiks/BetterDiscordApp/pull/288#issuecomment-271356216>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvNDDbioqlXvgV_4U19NIQI-IhNlykTks5rQnU3gaJpZM4INLQo>
> .
>

mkody commented:
Yes, it still works but you need to have my two PRs merged anyways (especially #436).

Until this is if merged upstream, I'd still recommend everyone to use my instructions and merge manually.
When Jiiks and co. will push updates to BD, we can just do a `git pull` and re-run the `install.sh`.

ddlr commented:
@mkody Thanks for sharing—I inserted `chmod u+x install.sh` before the last line and it worked like a charm so far. I’ll check if a BetterDiscord plugin I use still works in a moment.

simoniz0r commented:
Do I still have to follow mkody's instructions to install BetterDiscord?  I've tried cloning sarnex's updated PR and reinstalling BetterDiscord, but I still get the out of date error when launching Discord.

mkody commented:
Yup, #436 should fix it.

simoniz0r commented:
Any chance of all of that getting merged into this PR?  I'd rather not go through all of that every time I install BetterDiscord.

sarnex commented:
I merged #436 into my PR.

On Wed, Jan 11, 2017 at 11:33 AM, simoniz0r <notifications@github.com>
wrote:

> Any chance of all of that getting merged into this PR? I'd rather not go
> through all of that every time I install BetterDiscord.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Jiiks/BetterDiscordApp/pull/288#issuecomment-271918335>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvNDI9trwIK0AozKMbTuRu-9RSjJVp-ks5rRQRggaJpZM4INLQo>
> .
>

simoniz0r commented:
Thank you so much!

magiruuvelvet commented:
Still works with canary 0.0.14 :+1: 
I didn't tested the Linux PTB and Stable client. I prefer Canary :smile:

mkody commented:
I guess we should change the path in `js/main.js` as well.
```diff
diff --git a/js/main.js b/js/main.js
index 4ee8bea..3f1a5d9 100644
--- a/js/main.js
+++ b/js/main.js
@@ -13,7 +13,7 @@
     let __fs = window.require("fs");
     let __process = window.require("process");
     let __platform = __process.platform;
-    let __dataPath = (__platform === 'win32' ? __process.env.APPDATA : __platform === 'darwin' ? __process.env.HOME + '/Library/Preferences' : '/var/local') + '/BetterDiscord/';
+    let __dataPath = (__platform === 'win32' ? __process.env.APPDATA : __platform === 'darwin' ? __process.env.HOME + '/Library/Preferences' : process.env.HOME + '/.config') + '/BetterDiscord/';
 
 
     let __data = {};
```

simoniz0r commented:
Could you also please merge #435 into this PR?  I had to create a script that deletes `/home/user/.config/bdstorage.json` before launching BetterDiscord otherwise it complains that `/home/user/.config/BetterDiscord//bdStorage.json` doesn't exist, and gives me a javascript error that locks up the whole app.

![Javascript error](http://i.imgur.com/4jhuz2E.png)

sarnex commented:
I merged it.

On Thu, Jan 26, 2017 at 7:13 AM, simoniz0r <notifications@github.com> wrote:

> Could you also please merge #435
> <https://github.com/Jiiks/BetterDiscordApp/pull/435> into this PR? I had
> to create a script that deletes /home/user/.config/bdstorage.json before
> launching BetterDiscord otherwise it complains that /home/user/.config/
> BetterDiscord//bdStorage.json doesn't exist, and gives me a javascript
> error that locks up the whole app.
>
> [image: Javascript error]
> <https://camo.githubusercontent.com/e3d814ae6a97fc34c7d92b78e47782ba95ec6d5b/687474703a2f2f692e696d6775722e636f6d2f346a68757a32452e706e67>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Jiiks/BetterDiscordApp/pull/288#issuecomment-275374426>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvNDM6Y-qkJd1cWNiSM7eWiyQhmX5XFks5rWI3QgaJpZM4INLQo>
> .
>

magiruuvelvet commented:
And another update. Still works with 0.0.15 :+1:

ddlr commented:
@magiruuvelvet I hope you’re not going to keep this up every day, considering the new update schedule. 😜

TheDoctorsLife commented:
Canary it set to build on each new push to the master branch, just as an FYI.

simoniz0r commented:
I don't know if you've seen this, but Ckath made a script that can install BetterDiscord to any directory... might be of some use to you.
https://github.com/Ckath/Discord-stuff/blob/master/scripts/tape.sh

sarnex commented:
This is basically just doing what the BetterDiscord installer does in a
script, there's no way for it to be merged but maybe it's easier to use. I
recommend following the instructions here.

On Sat, Feb 4, 2017 at 8:52 AM, simoniz0r <notifications@github.com> wrote:

> I don't know if you've seen this, but Ckath made a script that can install
> BetterDiscord to any directory... might be of some use to you.
> https://github.com/Ckath/Discord-stuff/blob/master/scripts/tape.sh
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Jiiks/BetterDiscordApp/pull/288#issuecomment-277447245>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvNDNRzjwu6NDR1PvfPWzIPI4vPEXNdks5rZIKDgaJpZM4INLQo>
> .
>

simoniz0r commented:
I would say the fact that it allows any directory to be used is enough, but ¯\_(ツ)_/¯

sarnex commented:
It only has to be /opt for the install, you can move Discord after.

On Sat, Feb 4, 2017 at 1:56 PM, simoniz0r <notifications@github.com> wrote:

> I would say the fact that it allows any directory to be used is enough,
> but ¯_(ツ)_/¯
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Jiiks/BetterDiscordApp/pull/288#issuecomment-277466835>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvNDOwvjBv7-jHQKdglX6FAjwvhhK1gks5rZMnqgaJpZM4INLQo>
> .
>

simoniz0r commented:
Sure, but that's not ideal.  That is not the default install location for the .deb package.  Also, with PTB and Stable being released on Linux, `tape.sh` makes installing BetterDiscord to those Discord versions possible without having to move them to `/opt/DiscordCanary/` first.

Jiiks commented:
Maybe it's time to merge this o-o

mkody commented:
Woo!
You should keep an eye on the fact that we're not using `/var/local` anymore, but `~/.config`.
This includes (at least) in `v2/lib/core.js`: `installPath` and `dataPath`, and in`js/main.js`: for `__dataPath`

sarnex commented:
we did it boys, i dont mind maintaining the linux portion but please use
this PR for that

On Fri, Feb 10, 2017 at 9:07 AM, Kody <notifications@github.com> wrote:

> Woo!
> You should keep an eye on the fact that we're not using /var/local
> anymore, but ~/.config.
> This includes (at least) in v2/lib/core.js: installPath and dataPath, and
> injs/main.js: for __dataPath
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Jiiks/BetterDiscordApp/pull/288#issuecomment-278949781>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvNDNKtBTEQDtf4QGw-sDobvoFTkmb-ks5rbG8qgaJpZM4INLQo>
> .
>
